torch/_functorch/aot_autograd.py

```diff
@@ -2574,8 +2600,16 @@ def forward(ctx, *deduped_flat_tensor_args):
             num_mutated_metadata_only_inputs = (
                 CompiledFunction.metadata.num_mutated_metadata_only_inputs
             )
+            num_outputs_rng_offset = CompiledFunction.metadata.num_outputs_rng_offset
             # Our forward() returns both (mutated_inputs, outputs, output_intermediate_bases, saved_tensors, saved_symints)
             num_forward_returns = num_mutated_inputs + num_outputs + num_intermediate_bases
+            # In case of functionalization of rng ops, the fw_module returns one
+            # additinal output for rng offset. This rng offset is used right
+            # away to advance the rng state, and is not passed on to the raw
+            # outputs. However, we need to know the exact boundary to identify
+            # which tensors to be saved for the bwd graph.  num_forward captures
+            # this information.
+            num_forward = num_forward_returns + num_outputs_rng_offset
```

Comment:
@anijain2305 just curious - do you have an idea of whether export will have a need for functionalizing RNG? Is that something worth preemptively fitting inside of our export schema? (Instead of leaving it out, and having to add it later, I'm not sure what the BC risk there is).

Reply:
I think that some export users will want the functionalized RNG.

Reply:
@bdhirsh I am not sure yet. But I think we should assume that they will need functionalized RNG. So, let's review with that in mind.